src/mpid/ch4/src/ch4r_win.h

```diff
@@ -755,6 +755,10 @@ static inline int MPIDI_CH4R_win_finalize(MPIR_Win ** win_ptr)
             all_local_completed && all_remote_completed;
     } while (all_completed != 1);
 
+    mpi_errno = MPIDI_NM_mpi_win_free_hook(win);
+    if (mpi_errno != MPI_SUCCESS)
+        MPIR_ERR_POP(mpi_errno);
+
```

Comment:
Don't we also need to add shared memory hooks here?

Reply:
The SHM hooks will be introduced in a separate PR. I wanted to separate the original large SHM/RMA PR into small pieces.

Reply:
Ok, given that the shared memory RMA is pretty borked at this point, I guess it doesn't matter.